### PR TITLE
Surface untagged tracks in an explicit no-genre bucket

### PR DESCRIPTION
**Pull request by octaviospain**
## Summary

The genre index keyed audio items directly by their `genres` set, so a track with **no** genre tags landed in no bucket and was silently dropped from `genreIndexesProperty`. This change surfaces untagged tracks in a dedicated no-genre bucket.

A projection-only `Genre.None` sentinel is added to the sealed `Genre` hierarchy (empty name, mirroring the existing `Artist.UNKNOWN` unknown-entity convention; sorts first by natural name order). Both the core and JavaFX genre-index registries map an empty genre set to it via `it.genres.ifEmpty { setOf(Genre.None) }`. The bucket appears and disappears reactively as tracks gain or lose genres, and multi-key behaviour for tagged tracks is unchanged.

## Changes

**Production**
- `Genre.kt` — add `data object None : Genre("")`; inherits the existing name-based `compareTo` (no override). `Genre.Custom` now rejects an empty name, reserving `""` exclusively for `None` so a directly-constructed `Custom("")` cannot collide with the no-genre bucket in the name-ordered projection.
- `DefaultGenreIndexRegistry.kt` / `FXGenreIndexRegistry.kt` — key extractor maps an empty genre set to `setOf(Genre.None)`.
- `GenreIndexRegistryBase.kt` — `findFirst(genreName)` resolves a blank/empty argument to the `Genre.None` index.
- `AudioLibraryBase.kt` — `containsAudioItemWithGenre(genreName)` returns `true` for a blank argument when at least one untagged item exists.
- `ReactiveGenreIndex.kt` / `ImmutableGenreIndex.kt` — KDoc updated to describe the no-genre bucket.

`Genre.None` is a projection-only sentinel: it is never parsed, never persisted (`Genre` serialises as a comma-joined string and `None` never enters an item's `genres` set), and never registered in the known-genre lookup table.

**Tests**
- Core: untagged item lands in the `Genre.None` bucket; empty↔non-empty reactive transition; bucket disappears when the last untagged item is removed/tagged; blank-name query-helper resolution; `parseGenre` cannot produce `Custom("")`/`None`; `Custom("")` is rejected.
- JavaFX: the same no-genre bucket lifecycle (appear, transition, disappear) through `FXAudioLibrary`.
- Updated a persistence round-trip test to expect the `Genre.None` bucket for fixtures containing an untagged item.

## Verification

- [x] Full `gradle test` suite passes across all modules — no regressions.
- [x] `ktlintCheck` clean for all touched modules.
- [x] Code review pass: addressed a latent `Custom("")` collision hole, two stale KDoc statements, and a redundant cast.

Closes #164

### CODE REVIEW

File: music-commons-api/src/main/kotlin/net/transgressoft/commons/music/audio/Genre.kt [hunk 1]
```diff
@@ -34,8 +34,21 @@ sealed class Genre(open val name: String) : Comparable<Genre> {
...
+     * [None] is never parsed, never persisted, and never assigned to an item's `genres` set;
+     * it exists solely so the genre index can surface untagged tracks rather than dropping them.
+     */
+    data object None : Genre("")
+
     data class Custom(override val name: String) : Genre(name) {
         init {
+            // The empty name is reserved for the [None] sentinel: a Custom with an empty name would be
+            // compareTo-equal to None and collide with the no-genre bucket in the name-ordered projection.
+            require(name.isNotEmpty()) { "Custom genre name must not be empty" }
```
[the remark this answers is not in the document]
Comment (octaviospain) [reply music-commons-api/src/main/kotlin/net/transgressoft/commons/music/audio/Genre.kt]: Good catch — applied. Tightened the guard to `require(name.isNotBlank())` so a whitespace-only `Custom(" ")` is rejected too. As you noted, the blank-name query helpers (`findFirst`, `containsAudioItemWithGenre`) route blank arguments to `Genre.None`, so a whitespace-only custom genre would be lossy and unqueryable by name. Also extended the `GenreExtensions` test to cover the whitespace-only case.
